tests/integration/new_relations/test_new_relations.py

```diff
@@ -184,6 +194,35 @@ async def test_database_relation_with_charm_libraries(ops_test: OpsTest):
             cursor.execute("DROP TABLE test;")
 
 
+@pytest.mark.group("new_relations_tests")
+@pytest.mark.abort_on_fail
+async def test_filter_out_degraded_replicas(ops_test: OpsTest):
+    primary = await get_primary(ops_test, f"{DATABASE_APP_NAME}/0")
+    replica = next(
+        unit.name
+        for unit in ops_test.model.applications[DATABASE_APP_NAME].units
+        if unit.name != primary
+    )
+    machine = await get_machine_from_unit(ops_test, replica)
+    await stop_machine(ops_test, machine)
+
+    # Topology observer runs every half a minute
+    await asyncio.sleep(60)
+
+    for attempt in Retrying(stop=stop_after_attempt(3), wait=wait_fixed(3), reraise=True):
+        with attempt:
+            data = await get_application_relation_data(
+                ops_test,
+                APPLICATION_APP_NAME,
+                FIRST_DATABASE_RELATION_NAME,
+                "read-only-endpoints",
+            )
+            assert data is None
```

Comment:
There are two units in the cluster, so there will be no readonly endpoints if the down unit is filtered out.